File: activity/activity.go
package activity

import (
	"encoding/json"
	"errors"
	"fmt"
	"io/ioutil"
	"log"
)

var ActivityMap = map[string]string{
	"MOTORCYCLE": "./activity-schemas/motorbike.json",
	"HIKE": "./activity-schemas/hike.json",
	// "KAYAK": "./activity-schemas/kayak.json",
	// "RUN": "./activity-schemas/run.json",
	// "HORSE": "./activity-schemas/run.json",
}

func GetActivityByKey(t string) (activitySchema *ActivitySchema, err error) {
	activityFile, ok := ActivityMap[t]
	if !ok {
		return nil, errors.New("activity does not have a schema")
	}
	fmt.Println("Found activity file", activityFile)
	content, err := ioutil.ReadFile(activityFile)
	if err != nil {
		return nil, errors.New("activity schema not found")
	}

	// Now let's unmarshall the data into `payload`
	err = json.Unmarshal(content, &activitySchema)
	if err != nil {
		fmt.Println("Error during Unmarshal(): ", err.Error())
		// panic(err.Error())
		return
	}

	// Let's print the unmarshalled data!
	log.Printf("verb: %s\n", activitySchema.Verb)
	return
}

File: backend/fun_forecast_backend/activity_schema/activity.py
import json
from typing import Dict

import dacite

from .dataclasses import ActivitySchema


activity_map: Dict[str, str] = {
    "MOTORCYCLE": "../../resources/activity-schemas/motorbike.json",
    "HIKE": "../../resources/activity-schemas/hike.json",
    # "KAYAK": "../../resources/activity-schemas/kayak.json",
    # "RUN": "../../resources/activity-schemas/run.json",
    # "HORSE": "../../resources/activity-schemas/horse.json",
}


def get_activity_schema_by_key(key: str) -> ActivitySchema:
    if key not in activity_map:
        raise KeyError

        # add file exists check
    json_data = json.load(activity_map[key])
    return dacite.from_dict(ActivitySchema, json_data)
